cache/lfu.go

```diff
@@ -0,0 +1,127 @@
+// lfu.go
+// description: a type of cache algorithm used to manage memory within a computer.
+// details:
+// The standard characteristics of this method involve the system keeping track of the number of times a block is referenced in memory.
+// When the cache is full and requires more room the system will purge the item with the lowest reference frequency.
+// ref: (https://en.wikipedia.org/wiki/Least_frequently_used)
+// time complexity: O(N)
+// space complexity: O(1)
+// author: [CocaineCong](https://github.com/CocaineCong)
+
+package cache
+
+import (
+	"container/list"
+	"math"
+)
+
+// LFU the Least Frequently Used (LFU) page-replacement algorithm
+type LFU struct {
+	len     int // length
+	cap     int // capacity
+	minFreq int // The element that operates least frequently in LFU
+
+	// key: key of element, value: value of element
+	itemMap map[string]*list.Element
+
+	// key: frequency of possible occurrences of all elements in the itemMap
+	// value: elements with the same frequency
+	freqMap map[int]*list.List
+}
+
+// NewLFU init the LFU cache with capacity
+func NewLFU(capacity int) LFU {
+	return LFU{
+		len:     0,
+		cap:     capacity,
+		minFreq: math.MaxInt,
+		itemMap: make(map[string]*list.Element),
+		freqMap: make(map[int]*list.List),
+	}
+}
+
+// InitItem to insert item for LFU
+func InitItem(k string, v any, f int) item {
```

Comment:
This could really cause a lot of problems - you have an exported method that is returning an unexported type, meaning if you were to try to do anything with this `item`outside of this package, you would essentially have a `struct{}` that you have no control over. If that was by design, then doesn't it make sense to make this function unexported as well.

Reply:
sorry for my overlooking this detail. I fix it now.